### PR TITLE
Unit test for HandshakeInfo.Equal 

### PR DESCRIPTION
**Pull request by janardhanvissa**
Addresses: #6568 

RELEASE NOTES: None

### CODE REVIEW

**Comment by janardhanvissa** [comment]
@purnesh42H Added Unit test for Equal. Please review the changes and let me know any changes are required.

File: internal/credentials/xds/handshake_info_test.go [hunk 1]
```diff
@@ -300,3 +312,102 @@ func TestMatchingSANExists_Success(t *testing.T) {
...
+			},
+			wantMatch: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			gotMatch := test.hi1.Equal(test.hi2)
+			if gotMatch != test.wantMatch {
+				t.Errorf("wantMatch %v; hi1.Equal(hi2) = %v", test.wantMatch, gotMatch)
```
Comment (purnesh42H) [inline internal/credentials/xds/handshake_info_test.go:409]: @janardhanvissa please change to got before want convention

**Comment by janardhanvissa** [comment]
@purnesh42H Updated the error message in the t.Errorf() function to use the got before want convention.

**Comment by purnesh42H** [comment]
@janardhankrishna-sai create a new PR fresh from the latest main branch to avoid previous commits being shown

**Pull request closed without merging**